compiler/AST/build.cpp

```diff
@@ -1736,6 +1719,14 @@ buildClassDefExpr(const char* name,
                   Flag        isExtern,
                   const char* docs) {
   AggregateType* ct = toAggregateType(type);
+  if (strcmp("string", name) == 0) {
+    *dtString = *ct;
+    dtString->fields.parent = dtString;
+    dtString->inherits.parent = dtString;
```

Comment:
I've added a comment that talks about this. The constructor for an `AggregateType` sets

```c++
fields.parent = this;
inherits.parent = this;
```
The assignment `*dtString = *ct;` will replace these pointers in `dtString` with the address `ct`, so we need to go back and fix it up.